saleor/dashboard/product/views.py

```diff
@@ -80,10 +82,15 @@ def product_class_delete(request, pk):
                 'Dashboard message',
                 'Deleted product type %s') % product_class)
         return redirect('dashboard:product-class-list')
-    return TemplateResponse(
-        request,
-        'dashboard/product/product_class/modal_confirm_delete.html',
-        {'product_class': product_class, 'products': products})
+
+    products = [str(p) for p in product_class.products.all()]
+    context = {'product_class': product_class, 'products': products}
+    url_root = 'dashboard/product/product_class/'
+    if products:
```

Comment:
I don't think we need separate templates for this case. We could have one template with context variable `has_products` and if it's true show a warning with the number of products currently associated with the product class. Also I wouldn't forbid that action - if is user is aware of removing products maybe that's something that he's trying to accomplish.